main.rb

```diff
@@ -24,4 +27,31 @@ def can_use_services?
 
     true
   end
+
+  def correct_name
+    @name
+  end
+end
+
+class BaseDecorator < Nameable
+  def initialize(props)
+    super()
+    @nameable = props
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name[0, 10]
```

Comment:
As per the requirements, this function should check for name length, and if the length of the name > 10, should return a trimmed name of up to 10 characters otherwise should return the name as it is. ⬇️
![image](https://user-images.githubusercontent.com/33268927/215833600-2103a288-1371-48cf-95de-5b19926fe19c.png)